### PR TITLE
Apply editor.indent preference

### PR DESCRIPTION
**Pull request by facchinm**
Solves #6128

### CODE REVIEW

**Comment by bombasticbob** [comment]
tested the linux64 version.  it now honors the true/false state of the 'editor.indent' setting for indenting.  however, the _unwanted_ behavior of adding a '}' for me when I type '{' is still there.

Thank you for fixing this part of the issue.  I have noted that the '{' '}' "autocomplete" behavior is no longer directly associated with indenting. It happens whether or not indenting is enabled.

I shall add a new issue to use a 'preferences.txt' string to _*disable*_ the unwanted (irritating) behavior, and reference these issues as part of it.

**Comment by bombasticbob** [comment]
related to this issue, I found the code that inserts the ending curly brace, see latest comment added to #6132 (with recommended change)

File: app/src/processing/app/EditorTab.java [hunk 1]
```diff
@@ -148,6 +148,7 @@ private SketchTextArea createTextArea(RSyntaxDocument document)
     textArea.setMarginLineEnabled(false);
     textArea.setCodeFoldingEnabled(PreferencesData.getBoolean("editor.code_folding"));
     textArea.setAutoIndentEnabled(PreferencesData.getBoolean("editor.indent"));
+    textArea.setCloseCurlyBraces(PreferencesData.getBoolean("editor.indent"));
```
Comment (matthijskooijman) [inline app/src/processing/app/EditorTab.java]: Wouldn't this be a separate preference? I personally dislike autocompletion, but I do want (same-level) auto indent (i.e. pressing enter preserves the indentation of the previous line, it should not increase indentation after a { for me).

Comment (facchinm) [inline app/src/processing/app/EditorTab.java]: Agree, it should be moved to its own preference. I added it this way to check if the functionality was ok and provide a build with it

**Comment by bombasticbob** [comment]
@facchinm
@matthijskooijman

Thanks for the activity on this.  Yes, I'm in total agreement with your earlier comments about making these two functionalities separate.  Since the forum post from a couple of years ago (mentioned at the beginning) suggested that getting rid of auto-indent also got rid of auto-closing-brace, I had to weigh getting rid of something I want (auto-indent) against getting rid of something I _really_ do _not_ want (auto-closing-brace).  Having the two settings instead (i.e. #6132) would be the best overall fix for this.

So thanks for the mention.  Also thanks for testing how it _can_ be done.

**Comment by bombasticbob** [comment]
Here's a patch file that I have created based on a 'git clone' I did on the repository.  It includes the change that fixes 'indent' (noted above) as well as adding 'editor.autocomplete' (issue #6132).  I tested it and it works perfectly.  It requires manually editing the local 'preferences.txt' to turn OFF the feature, but that's fine with me and probably everyone who wants this feature on by default.


[issue_6132.patch.txt](https://github.com/arduino/Arduino/files/896350/issue_6132.patch.txt)

**Comment by bombasticbob** [comment]
@facchinm 

please incorporate #6155 as well.  then at least 3 additional people would be happier (and most likely _many_ others).

**Comment by cmaglie** [comment]
@bombasticbob 
the option `editor.autocomplete`, as it is now, disables "auto close braces" but not "auto indent", while I expect it to override both (even if `editor.indent` is true).

Anyway, the editing-aids are just "auto indent" and "auto close braces" and IMHO is too early to think about an option to disable the aids altogether, I'd add such option later when/if more editing-aids are added.

So here my proposal:

- Change `editor.autocomplete` (that seems too generic to me) to `editor.auto_close_braces` and merge this PR as is
- Postpone the option to disable all editing-aids to a future PR when more editing-aids are added

**Comment by bombasticbob** [comment]
@cmaglie 

changing the text of the setting to 'auto_close_braces' is ok with me.  I just want a way to control the behavior.  I'm happy that the only question is what actual name to use [something I had already guessed might happen].

**Pull request merged**